### PR TITLE
gh-111482: Use Argument Clinic for clock_gettime()

### PR DESCRIPTION
**Pull request by vstinner**
Use Argument Clinic for time.clock_gettime() and
time.clock_gettime_ns() functions.

Benchmark on time.clock_gettime_ns():

    import time
    import pyperf
    runner = pyperf.Runner()
    runner.timeit(
        'clock_gettime_ns(CLOCK_MONOTONIC_COARSE)',
        setup='import time; clock_gettime_ns=time.clock_gettime_ns; CLOCK_MONOTONIC_COARSE=6',
        stmt='clock_gettime_ns(CLOCK_MONOTONIC_COARSE)')

Result on Linux with CPU isolation:

Mean +- std dev: [ref] 134 ns +- 1 ns -> [change] 55.7 ns +- 1.4 ns: 2.41x faster

* Issue: gh-111482

### CODE REVIEW

**Comment by vstinner** [comment]
cc @corona10

**Comment by corona10** [comment]
@vstinner 
Please run the following command :)

````
make regen-all 
````

**Review by corona10 — APPROVED** [review]
Overall LGTM

File: Modules/_testinternalcapi.c [hunk 1]
```diff
@@ -1639,6 +1639,44 @@ perf_trampoline_set_persist_after_fork(PyObject *self, PyObject *args)
 }
 
 
+static PyObject *
+bench_asutf8(PyObject *self, PyObject *args)
```
Comment (serhiy-storchaka) [inline Modules/_testinternalcapi.c]: I think it is unrelated.

**Review by serhiy-storchaka — COMMENTED** [review]
Wow, it is impressive! I did not know there are still such low hanging fruits.

This PR consists of two parts.

1. `clock_gettime()` supported 64-bit `clockid_t` on AIX. This PR makes `clock_gettime_ns()` also supporting it. It looks rather like a bugfix, and I think it should be backported. `clock_settime()` and `clock_settime_ns()` also should support it.
2. Using Argument Clinic instead of `PyArg_ParseTuple()`.

File: Modules/_testinternalcapi.c (hunk 1 above)
Comment (vstinner) [reply Modules/_testinternalcapi.c]: Ooops, reverted.

**Comment by vstinner** [comment]
> Wow, it is impressive! I did not know there are still such low hanging fruits.

Honestly, I'm disappointed that PyArg_ParseTuple() is so slow :-( I suppose that creating a temporary tuple to pass positional arguments also has a cost. But yeah, it's a nice optimization.

> clock_gettime() supported 64-bit clockid_t on AIX. This PR makes clock_gettime_ns() also supporting it. It looks rather like a bugfix, and I think it should be backported. clock_settime() and clock_settime_ns() also should support it.

We don't support AIX. I just tried to not regress. I don't plan to backport Argument Clinic code nor enhance time.clock_gettime_ns() on AIX in 3.12.

**Comment by vstinner** [comment]
Thanks for reviews.

@serhiy-storchaka: If you are motivated to enhance AIX support in 3.12, please go ahead ;-) The two AIX buildbots are **failing** for **many months**... I'm not even sure why we still have AIX buildbots.

**Pull request merged**